src/Codec.elm

```diff
@@ -442,6 +452,54 @@ nullableField name getter codec ocodec =
     field name getter (maybe codec) ocodec
 
 
+{-| Basically a `maybeField` where data is validated when provided and a default value to use when not.
```

Comment:
How about:
```
{- | Similar to the `maybeField` codec, with added validation when the data exists. This means that bad data will
    generate a decoding error, instead of silently failing and returning a `Nothing`.

    - valid data provided: returns `Just` the decoded data
    - no data provided (either the field is missing, or it is null): returns `Nothing`
    - invalid data provided: returns a decoding error

    type alias JustAnInt = { myInt: Maybe Int }

    objectCodec = Codec.object JustAnInt
        |> Codec.optionalField "myInt" .myInt Codec.int
        |> Codec.buildObject

    Codec.decodeString objectCodec "{\"myInt\": \"Not an int\"}"
    --Returns a json decoding error with the message containing: "Expecting an INT"
-}
```